frog/imports/api/activities.js

```diff
@@ -18,13 +24,18 @@ export const Results = new Mongo.Collection('results');
 
 export const getInstances = (
   activityId: string
-): [?(string[]), structureDefT] => {
+): { groups: string[], structure: structureDefT } => {
```

Comment:
I agree, this is clearer. It was fine while I was using const [groups, structure] = getInstances, but when I saw you using getInstances[1], I realized that it was very hard to understand.